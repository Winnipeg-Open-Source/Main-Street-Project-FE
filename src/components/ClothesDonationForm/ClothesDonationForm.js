import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Input from 'components/Input';
import Select from 'components/Select';
import LineItemForm from 'components/LineItemForm';
import TextArea from 'components/TextArea';

function ClothesDonationForm ({
    name,
    description,
    size,
    locations,
    lineItems,
    notes,
    onChange,
    onAddLineItemClick,
    onItemQuantityChange,
}) {
    return (
        <>
            <Input id='name' name='name' placeholder='Name' value={name} onChange={onChange} />
            <Input id='description' name='description' placeholder='Description' value={description} onChange={onChange} />
            <Select id='size' name='size' label='Size' value={size} onChange={onChange} data-testid='Size'>
                <option>XS</option>
                <option>S</option>
                <option>M</option>
                <option>L</option>
                <option>XL</option>
                <option>XXL</option>
                <option>XXXL</option>
            </Select>
            <LineItemForm
                locations={locations}
                lineItems={lineItems}
                onAddClick={onAddLineItemClick}
                onItemQuantityChange={onItemQuantityChange}
            />
            <TextArea id='notes' name='notes' label='Notes' value={notes} onChange={onChange} />
        </>
    );
}

ClothesDonationForm.displayName = 'ClothesDonationForm';

ClothesDonationForm.propTypes = {
    name: PropTypes.string,
    description: PropTypes.string,
    size: PropTypes.string,
    quantity: PropTypes.number,
    notes: PropTypes.string,
    onChange: PropTypes.func,
    onAddLineItemClick: PropTypes.func,
    onItemQuantityChange: PropTypes.func,
};

ClothesDonationForm.defaultProps = {
    size: 'M'
};

export default memo(ClothesDonationForm);
