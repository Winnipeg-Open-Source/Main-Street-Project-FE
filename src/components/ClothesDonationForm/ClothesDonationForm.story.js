import React from 'react';
import { action } from '@storybook/addon-actions';
import { Flex } from 'pcln-design-system';
import ClothesDonationForm from './ClothesDonationForm';

const props = {
    name: 'T-Shirt',
    description: 'Pretty Cool T-Shirt',
    size: 'L',
    quantity: 5,
    notes: 'A bunch of notes',
    onNameChange: action('name change'),
    onDescriptionChange: action('description change'),
    onSizeChange: action('size change'),
    onQuantityChange: action('quantity change'),
    onNotesChange: action('notes change'),
};

export const EmptyForm = () => (
    <Flex p={2}>
        <ClothesDonationForm />
    </Flex>
);

export const CompletedForm = () => (
    <Flex p={2}>
        <ClothesDonationForm {...props} />
    </Flex>
);

export default {
    title: 'ClothesDonationForm',
    component: ClothesDonationForm,
};
