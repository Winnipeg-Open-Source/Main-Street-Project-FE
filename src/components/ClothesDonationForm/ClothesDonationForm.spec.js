import React from 'react';
import { render } from '@testing-library/react';
import ClothesDonationForm from './ClothesDonationForm';

const props = {
    name: 'T-Shirt',
    description: 'Pretty Cool T-Shirt',
    size: 'L',
    quantity: 5,
    notes: 'A bunch of notes',
    onNameChange: () => {},
    onDescriptionChange: () => {},
    onSizeChange: () => {},
    onQuantityChange: () => {},
    onNotesChange: () => {},
};

describe ('ClothesDonationForm', () => {
    it ('renders without crashing', () => {
        const { asFragment } = render(<ClothesDonationForm />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('has placeholders when empty', async () => {
        const { getByLabelText, getByPlaceholderText, getByTestId } = render(<ClothesDonationForm />);

        const name = await getByPlaceholderText('Name');
        const description = await getByPlaceholderText('Description');
        const size = await getByTestId('Size');
        const quantity = await getByPlaceholderText('Quantity');
        const notes = await getByLabelText('Notes');

        expect(name.value).toBe('');
        expect(description.value).toBe('');
        expect(size.value).toBe('M');
        expect(quantity.value).toBe('');
        expect(notes.value).toBe('');
    });

    it ('has all fields used', async () => {
        const { getByLabelText, getByPlaceholderText, getByTestId } = render(<ClothesDonationForm {...props} />);

        const name = await getByPlaceholderText('Name');
        const description = await getByPlaceholderText('Description');
        const size = await getByTestId('Size');
        const quantity = await getByPlaceholderText('Quantity');
        const notes = await getByLabelText('Notes');

        expect(name.value).toBe('T-Shirt');
        expect(description.value).toBe('Pretty Cool T-Shirt');
        expect(size.value).toBe('L');
        expect(quantity.value).toBe('5');
        expect(notes.value).toBe('A bunch of notes');
    });
});
