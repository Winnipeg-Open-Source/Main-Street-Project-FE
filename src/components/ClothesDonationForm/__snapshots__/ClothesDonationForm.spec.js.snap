// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClothesDonationForm renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pkfPC haHNKR sc-psDhf iAENMo"
  >
    <input
      class="sc-qcrrk sc-qQxFs iUvCRq"
      id="name"
      name="name"
      placeholder="Name"
      value=""
    />
    <input
      class="sc-qcrrk sc-qQxFs iUvCRq"
      id="description"
      name="description"
      placeholder="Description"
      value=""
    />
    <input
      class="sc-qcrrk sc-qQxFs iUvCRq"
      id="quantity-name"
      name="quantityName"
      placeholder="Quantity Name"
      value=""
    />
    <label
      class="sc-pdKru dQCRbh className"
      color="border.light"
      font-size="10px"
      font-weight="bold"
    >
      Size
      <div
        class="sc-AxirZ sc-AxhCb czMtkd"
        width="1"
      >
        <select
          class="sc-oTaid bpOcmz sc-paWVw sc-qYUWV imGcog"
          data-testid="Size"
          font-size="2,,1"
          id="size"
          name="size"
        >
          <option>
            XS
          </option>
          <option>
            S
          </option>
          <option>
            M
          </option>
          <option>
            L
          </option>
          <option>
            XL
          </option>
          <option>
            XXL
          </option>
          <option>
            XXXL
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="sc-AxmLO fTwYWQ sc-fzoxKX sc-pZCuu bTxHcb"
          color="text.light"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
          />
        </svg>
      </div>
    </label>
    <div
      class="sc-AxirZ sc-AxhCb sc-pjumZ jtOhCp"
      width="1"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        role="combobox"
        style="position: relative; width: 100%;"
      >
        <input
          aria-autocomplete="list"
          aria-labelledby="downshift-0-label"
          autocomplete="off"
          class="sc-qcrrk gTabPy sc-pRgDJ bNviUc"
          font-size="2,,1"
          id="downshift-0-input"
          placeholder="Location Search"
          value=""
          width="1"
        />
        <div
          aria-labelledby="downshift-0-label"
          class="sc-AxirZ sc-pCPXO sc-pBlxj gmOuQA"
          height="256px"
          id="downshift-0-menu"
          role="listbox"
        />
      </div>
    </div>
    <label
      class="sc-pdKru dQCRbh"
      color="border.light"
      font-size="10px"
      font-weight="bold"
    >
      Notes
      <textarea
        class="sc-pAYXY sc-qQKeD gqDCoc"
        id="notes"
        name="notes"
      />
    </label>
  </div>
</DocumentFragment>
`;
